src/metkit/api/metkit_c.cc

```diff
@@ -0,0 +1,348 @@
+#include "metkit_c.h"
+#include "metkit/mars/MarsExpension.h"
+#include "metkit/mars/MarsRequest.h"
+#include "metkit/metkit_version.h"
+
+#include "eckit/exception/Exceptions.h"
+#include "eckit/runtime/Main.h"
+#include "eckit/utils/Optional.h"
+
+#include <functional>
+
+extern "C" {
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+struct metkit_marsrequest_t : public metkit::mars::MarsRequest {
+    using metkit::mars::MarsRequest::MarsRequest;
+
+    metkit_marsrequest_t(metkit::mars::MarsRequest&& k) :
+        metkit::mars::MarsRequest(std::move(k)) {}
+};
+
+struct metkit_requestiterator_t {
+    metkit_requestiterator_t(std::vector<metkit::mars::MarsRequest>&& vec) :
+        vector_(std::move(vec)), iterator_(vector_.begin()) {}
+
+    int next() {
+        if (iterator_ == vector_.end()) {
+            return METKIT_ITERATION_COMPLETE;
+        }
+        ++iterator_;
+        return iterator_ == vector_.end() ? METKIT_ITERATION_COMPLETE : METKIT_SUCCESS;
+    }
+
+    void current(metkit_marsrequest_t* request)  {
+        ASSERT(iterator_ != vector_.end());
+        *request = std::move(*iterator_);
+    }
+
+private:
+
+    std::vector<metkit::mars::MarsRequest> vector_;
+    std::vector<metkit::mars::MarsRequest>::iterator iterator_;
+};
+
+/// @comment: (maby)
+/// Not sure if there is much value in having a param iterator. We could just return an array of
+/// strings (char**) using metkit_marsrequest_params.
+/// I think we should metkit_paramiterator_t.
```

Comment:
Here the comment is lacking some words. Do this need to be addressed before merging?

Reply:
a todo left for me by me